tools/src/wyvern/tools/tests/ILTests.java

```diff
@@ -136,4 +159,58 @@ public void testIdentityCall() throws ParseException {
     	IntegerLiteral five = new IntegerLiteral(5);
 		Assert.assertEquals(five, v);
 	}
+    
+	@Test
+	public void testSingleModule() throws ParseException {
+		
+		String source = TestUtil.readFile(PATH + "example.wyv");
+		TypedAST ast = TestUtil.getNewAST(source);
+		
+		GenContext genCtx = GenContext.empty().extend("system", new Variable("system"), new NominalType("", "system")).extend("D",  new Variable("D"), new NominalType("", "D"));
+		wyvern.target.corewyvernIL.decl.Declaration decl = ((Declaration) ast).topLevelGen(genCtx);
+	}
+	@Test
+	public void testMultipleModules() throws ParseException {
+		
+		String[] fileList = {"A.wyt", "B.wyt", "D.wyt", "A.wyv", "D.wyv", "B.wyv", "main.wyv"};
+		GenContext genCtx = GenContext.empty().extend("system", new Variable("system"), new NominalType("", "system"));
+		
+		for(String fileName : fileList) {
```

Comment:
This looks like generic module linking code.  Should this be somewhere in the main codebase rather than embedded in a test?